torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -3036,6 +3038,39 @@ void TensorExprKernel::bindConstant(const torch::jit::Value* v) {
   bufs_[v] = buf;
 }
 
+void* allocBuf(size_t size) {
+  void* bp = (void*)malloc(size);
+  return bp;
+}
+
+void TensorExprKernel::preAllocIntermediateBufs(
+    const std::unordered_set<BufPtr> intem_bufs) {
+  void* bp;
+  for (auto buf : intem_bufs) {
+    // Check if buf shape is static and compute its size if static.
+    bool is_static = true;
+    size_t size =
+        elementSize(buf->dtype().scalar_type()) * buf->dtype().lanes();
+    for (auto d : buf->dims()) {
+      if (!d->isConstant()) {
+        is_static = false;
+        break;
+      }
+      size = size * immediateAs<int>(d);
```

Comment:
With @bertmaher's #64077 it might be better to use `immLike` function here so that it works for 64-bit ints as well (dimensions might be 64-bit).